spec/helpers/application_helper/buttons/middleware_standalone_server_action_spec.rb

```diff
@@ -1,8 +1,21 @@
 describe ApplicationHelper::Button::MiddlewareStandaloneServerAction do
-  let(:record) { double("MiddlewareServer") }
+  let(:record) { FactoryGirl.create(:middleware_server) }
   subject(:action) { described_class.new(setup_view_context_with_sandbox({}), {}, {'record' => record}, {}) }
 
+  let(:datasource_record) { FactoryGirl.create(:middleware_datasource, :middleware_server => record) }
+  let(:deployment_record) { FactoryGirl.create(:middleware_deployment, :middleware_server => record) }
+
   describe '#visible?' do
+    %i(datasource_record deployment_record).each do |delegated_class|
+      it "is delegated for #{delegated_class}" do
+        action = described_class.new(setup_view_context_with_sandbox({}), {}, {'record' => send(delegated_class)}, {})
+        allow(record).to receive(:in_domain?) { false }
+        allow(record).to receive(:mutable?) { true }
+
+        expect(action).to be_visible
```

Comment:
If you have `subject` instead of `action`, you can do `is_expected.to be_visible`